DaisyMicheal/component_library
# Feature/form component

**What does this PR do?**
Create a Form component library to handle user input and submission.

**Description of Task to be completed:**

- Create a Form component with customizable input fields and a submit button.
- Implement validation for required fields and email format.
- Ensure the form is accessible, including keyboard navigation and screen-reader support.
- Add customizable props such as onSubmit, onChange, and initial form values.
- Write unit tests for form validation, submission, and error states.
- Create a Storybook story to demonstrate various states and props of the form component.

**How should this be manually tested?**

1. Pull the branch and install dependencies if necessary.
2. Import the Form component into a test file or page.
3. Test the form with various input values and props to ensure it works as expected.
4. Verify that the form component is accessible and functions correctly in all scenarios.
5. Run the unit tests to ensure they pass.

**Any background context you want to provide?**
None

**What are the relevant pivotal tracker stories?**
[#188039425](https://www.pivotaltracker.com/story/show/188039425)

**Screenshots (if appropriate):**
![Screenshot from 2024-07-30 23-51-13](https://github.com/user-attachments/assets/4232d392-43d1-4494-a7fe-e10ea3576d81)
![Screenshot from 2024-07-30 23-51-59](https://github.com/user-attachments/assets/854b2972-0fd1-420c-9dd1-14474f829704)


**Questions:**
None
